openwisp_ipam/tests/test_multitenant.py

```diff
@@ -48,3 +50,222 @@ def test_multitenancy_subnet_queryset(self):
+        )
+
+    def test_subnet(self):
+        org_a = self._get_org(org_name='org_a')
+        self._login(username='user_a', password='tester')
+
+        # Subnet to be accessible by superusers and org_a users
+        subnet = self._create_subnet(subnet='10.0.0.0/24', organization=org_a)
+        response = self.client.get(reverse('ipam:subnet', args=(subnet.id,)))
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data['id'], str(subnet.id))
+        self._login(username='superuser', password='tester')
+        response = self.client.get(reverse('ipam:subnet', args=(subnet.id,)))
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data['id'], str(subnet.id))
+
+        # Should throw PermissionError for users from other organizations
+        self._login(username='user_b', password='tester')
+        response = self.client.get(reverse('ipam:subnet', args=(subnet.id,)))
+        self.assertEqual(response.status_code, 403)
+
+    def test_subnet_hosts(self):
+        org_a = self._get_org(org_name='org_a')
+        subnet = self._create_subnet(subnet='10.0.0.0/24', organization=org_a)
+        self._login(username='user_a', password='tester')
+        response = self.client.get(reverse('ipam:hosts', args=(subnet.id,)))
+        self.assertEqual(response.status_code, 200)
+        self._login(username='superuser', password='tester')
+        response = self.client.get(reverse('ipam:hosts', args=(subnet.id,)))
+        self.assertEqual(response.status_code, 200)
+        self._login(username='user_b', password='tester')
+        response = self.client.get(reverse('ipam:hosts', args=(subnet.id,)))
+        self.assertEqual(response.status_code, 404)
+
+    def test_subnet_list_ipaddress(self):
+        org_a = self._get_org(org_name='org_a')
+        subnet = self._create_subnet(subnet='10.0.0.0/24', organization=org_a)
+        self._login(username='superuser', password='tester')
+        post_data = self._post_data(ip_address='10.0.0.5', subnet=str(subnet.id))
+        response = self.client.post(
+            reverse('ipam:list_create_ip_address', args=(subnet.id,)),
+            data=post_data,
+            content_type='application/json',
+        )
+        self.assertEqual(response.status_code, 201)
+        response = self.client.get(
+            reverse('ipam:list_create_ip_address', args=(subnet.id,))
+        )
+        self.assertEqual(response.status_code, 200)
+        self._login(username='user_a', password='tester')
+        response = self.client.get(
+            reverse('ipam:list_create_ip_address', args=(subnet.id,))
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data['results'][0]['ip_address'], '10.0.0.5')
+        self._login(username='user_b', password='tester')
+        response = self.client.get(
+            reverse('ipam:list_create_ip_address', args=(subnet.id,))
+        )
```

Comment:
there's quite some assertions here, can you please add a few subtests with a short description here to make it faster to read?